src/content.js

```diff
@@ -22,6 +22,43 @@ export function tidyHtml (element) {
   normalizeTags(element)
 }
 
+export function getPlainText (element) {
+  if (element.nodeType === nodeType.textNode) {
+    return element.textContent
+  }
+
+  if (element.nodeType === nodeType.elementNode) {
+    let result = ''
+
+    for (const childNode of element.childNodes) {
+      let childText = getPlainText(childNode)
+        .replace(/\s*\n+\s*/g, ' ') // collapse newlines within text
+
+      // collapse leading whitespace
+      if (/^\s+\S+/.test(childText)) {
```

Comment:
Maybe we should separate the whitespace normalisation from the tag filtering? Also we have some existing methods for whitespace filtering both in `string.js` as well as in this file. E.g. `removeWhitespaces()`. A thing to note is that when removing whitespaces it can leave empty text nodes which can make the whole thing a bit more complicated if one wants to leave the text nodes in a clean state after cleaning. If memory serves right this is what `removeWhitespaces()` does for trimming in the start and end.